code/obj/machinery/manufacturer.dm

```diff
@@ -26,62 +27,93 @@ TYPEINFO(/obj/machinery/manufacturer)
-	var/free_resource_amt = 0 //! The amount of each free resource that the manufacturer comes preloaded with
-	var/list/obj/item/material_piece/free_resources = list() //! See free_resource_amt; this is the list of resources being populated from
-	var/obj/item/disk/data/floppy/manudrive/manudrive = null
+	/* Resources/materials */
+	/// Base class for material pieces that the manufacturer accepts.
+	/// Keep this as material pieces only unless you're making larger changes to the system,
+	/// Various parts of the code are coupled to assuming that this is a material piece w/ a material.
+	var/base_material_class = /obj/item/material_piece
+	/// The amount of each free resource that the manufacturer comes preloaded with.
+	/// Separate from free_resources() as typically manufacturers use the same amount of each type.
+	var/free_resource_amt = 0
+	/// The types of material pieces of which the manufacturer will be spawned with.
+	/// The amount of each resource is defined on free_resource_amt
+	var/list/obj/item/material_piece/free_resources = list()
+	/// Supposedly used by podwards manufacturers, but not really
 	var/list/resource_amounts = list()
-	var/list/materials_in_use = list()
+	var/obj/item/disk/data/floppy/manudrive/manudrive = null
 	var/should_update_static = TRUE //! true by default to update first time around, set to true whenever something is done that invalidates static data
-	var/list/material_patterns_by_ref = list() //! Helper list which stores all the material patterns each loaded material satisfies, by ref to the piece
+	/// Helper list which stores all the material patterns each loaded material satisfies, by ref to the piece
+	var/list/material_patterns_by_ref = list()
 
-	// Production options
-	var/search = null
-	var/category = null
+	/* Production options */
+	/// A list of valid categories the manufacturer will use. Any invalid provided categories are assigned "Miscellaneous".
 	var/list/categories = list("Tool", "Clothing", "Resource", "Component", "Machinery", "Medicine", "Miscellaneous", "Downloaded")
 	var/accept_blueprints = TRUE
-	var/list/available = list() //! A list of every option available in this unit subtype by default
-	var/list/download = list() //! Options gained from scanned blueprints
-	var/list/drive_recipes = list() //! Options provided by an inserted manudrive
-	var/list/hidden = list() //! These options are available by default, but can't be printed unless the machine is hacked
+	/// A list of every option available in this unit subtype by default
+	var/list/available = list()
+	/// Options gained from scanned blueprints
+	var/list/download = list()
+	/// Options provided by an inserted manudrive
+	var/list/drive_recipes = list()
+	/// These options are available by default, but can't be printed or seen unless the machine is hacked
+	var/list/hidden = list()
```

Comment:
Broadly not a fan of the "option" noun in the comments here. I personally know what these vars are for, but I've been staring at manufacturer code this past month. These are lists of types, right?

Reply:
List of types yes. I'll update it accordingly